csharp/src/Google.Protobuf/WritingPrimitives.cs

```diff
@@ -189,9 +222,13 @@ public static void WriteString(ref Span<byte> buffer, ref WriterInternalState st
                     unsafe
                     {
                         fixed (char* sourceChars = &MemoryMarshal.GetReference(source))
-                        fixed (byte* destinationBytes = &MemoryMarshal.GetReference(buffer.Slice(state.position)))
+                        fixed (byte* destinationBytes = &MemoryMarshal.GetReference(buffer))
```

Comment:
It has a tiny effect on performance. There is no need to do a slice because we can jump to the right place with GetBytes using `destinationBytes + state.position`.